### PR TITLE
[tests][scripts][release] Optimize release note script about categories etc

### PR DESCRIPTION
**Pull request by ysh329**
* Optimize release note script generate process
  * `gahter_prs.py`: create one csv file only, keep sheet header name with next step script `make_notes.py` same;
  * `make_notes.py`: create a categories/tags dictionary, used to map various categories/tags to normal ones;
  * `README.md`: modifiction aligned with updated scripts etc.
* Fix typo in `release_process.rst`.

cc @driazati @Johnson9009

### CODE REVIEW

**Review by junrushao — APPROVED** [review]
Very cool. Thanks!

File: tests/scripts/release/README.md [hunk 1]
```diff
@@ -24,11 +24,13 @@ export GITHUB_TOKEN=<github oauth token>
 python release/gather_prs.py --from-commit $(git rev-parse v0.9.0) --to-commit $(git merge-base origin/main v0.8.0)
 ```
 
-You can then import this CSV into a collaborative spreadsheet editor to distribute the work of categorizing PRs for the notes. Once done, you can download the resulting CSV and convert it to readable release notes.
+After commands above, you will get a csv file named `out_pr_gathered.csv`. You can then import this CSV into a collaborative spreadsheet editor to distribute the work of categorizing PRs for the notes, **espeicially check the `pr_title_tags` column for each row and correct it if wrong**.
```
Comment (driazati) [inline tests/scripts/release/README.md:27]:
```suggestion
After running the commands above, you will get a csv file named `out_pr_gathered.csv`. You can then import this CSV into a collaborative spreadsheet editor to distribute the work of categorizing PRs for the notes, **especially check the `pr_title_tags` column for each row and correct it if it's wrong**.
```

File: tests/scripts/release/make_notes.py [hunk 2]
```diff
@@ -17,19 +17,81 @@
...
 from collections import defaultdict
-from typing import Callable, Dict, List, Any
 
 REPO_ROOT = Path(__file__).resolve().parent.parent.parent.parent
 sys.path.append(str(REPO_ROOT / "tests" / "scripts"))
 sys.path.append(str(REPO_ROOT / "tests" / "scripts" / "github"))
 sys.path.append(str(REPO_ROOT / "tests" / "scripts" / "jenkins"))
 
+# Tag dictionary used to create a mapping relation to categorize PRs owning same tag.
+TAG_DICT = {
```
Comment (driazati) [inline tests/scripts/release/make_notes.py:32]: Thanks for coming up with this list! It'll make auto-generating the notes much simpler

**Review by driazati — APPROVED** [review]

File: tests/scripts/release/make_notes.py (hunk 2 above)
Comment (ysh329) [reply tests/scripts/release/make_notes.py:32]: Yeah, this list will keep maintained gradualy, while monthly reports generation.

**Pull request merged**